### DISCUSSION TITLE
Using a custom plugin built on babel on a project running Nextjs

### DISCUSSION THREAD

**Discussion by luispcosta** [swc-project/swc#8719] [General]
Hello everyone. 

I'm not sure if this is the correct place to ask this question, so feel free to tell me where I should ask this!

I'm working on a team that is maintaining a product written in NextJS, and there's a plugin we'd really like to add to the product called [Annotate React](https://help.fullstory.com/hc/en-us/articles/360049493054-FullStory-s-Annotate-React-plugin-for-Web-Native). 

However this plugin was built to work with babel, which makes it impossible for us to use because we have many other packages that depend on SWC (we could use custom plugins for next as written [here](https://nextjs.org/docs/pages/building-your-application/configuring/babel), however doing that will automatically tell Next to switch from SWC to babel for the compilation process).

How could we go about writing some form of a wrapper for that plugin for SWC? Is there any resources I can check out to understand what needs to be done?

**Comment by kdy1** [comment]
You can't _wrap_ it.
You need to port it to a Rust-based Wasm plugin
